src/components/Modals/QrCode/Form.tsx

```diff
@@ -88,6 +95,10 @@ export const Form: React.FC<QrCodeFormProps> = ({ accountId }) => {
   const handleQrSuccess = useCallback(
     (decodedText: string) => {
       ;(async () => {
+        // If this is a WalletConnect dApp QR Code, skip the whole send logic and render the QR Code Modal instead.
+        // There's no need for any RFC-3986 decoding here since we don't really care about parsing and WC will do that for us
+        if (decodedText.startsWith('wc:')) return setWalletConnectDappUrl(decodedText)
```

Comment:
will this work with v2?

Reply:
Yeah they use the same `wc:` URN scheme as per [EIP-1328](https://eips.ethereum.org/EIPS/eip-1328), only the `version` param differs. I have tested with a bunch of different dApps (Hop, 1Inch, Safe, Uniswap), and just re-tested as a sanity check with https://react-dapp-v2-with-web3js.vercel.app/

<img width="224" alt="image" src="https://user-images.githubusercontent.com/17035424/236013108-bab4d601-24f3-415d-be62-53bd35f3dfe4.png">